src/helper/service.hpp

```diff
@@ -68,10 +70,21 @@ class service {
         return service_config_;
     }
 
+    std::optional<sampler::scope> schema_extraction_sampled()
```

Comment:
This should be a `get_schema_sampler` function returning a `shared_ptr<schema_sampler>` instead of having this helper here, for consistency.

Reply:
Done